### PR TITLE
Fix for issue #1554 + missing API integration tests

### PR DESCRIPTION
**Pull request by TjadenFroyda**
This PR:
- Fixes issue #1554 and updates associated unit tests.
- Adds integration tests for API endpoints, noted to be missing in #1554.
- Added CreateStratisPowApiNode to facilitate testing API functions with transactions/mining. 
- Adds an exception handler to FileStorage. This [unhandled exception](https://msdn.microsoft.com/en-us/library/system.io.file.delete(v=vs.110).aspx) was thrown when SaveWallet was attempting to delete a file.

All added integration test pass. Some unrelated tests (Wallet integration tests, mostly) not touched by this PR are failing when I run locally. 

image

### CODE REVIEW

File: src/Stratis.Bitcoin/Utilities/FileStorage.cs [hunk 1]
```diff
@@ -78,6 +78,10 @@ public void SaveToFile(T toSave, string fileName, bool saveBackupFile = false)
                 // Marking the file for deletion in the future.
                 File.Move(tempFilePath, $"{ filePath}.{ uniqueId}.del");
             }
+            catch (UnauthorizedAccessException)
```
Comment (dangershony) [inline src/Stratis.Bitcoin/Utilities/FileStorage.cs]: You don't want to swallow the exception here.

Comment (TjadenFroyda) [reply src/Stratis.Bitcoin/Utilities/FileStorage.cs]: Fixed

**Comment by dangershony** [comment]
Absolutely brilliant @TjadenFroyda

File: src/Stratis.Bitcoin.IntegrationTests/API/ApiSpecification.cs [hunk 2]
```diff
@@ -35,5 +35,127 @@ public void Listmethods_via_rpc_returns_non_empty_list()
...
+            And(the_block_should_contain_the_transaction);
+        }
+
+        [Fact]
+        public void Getblockcount_via_api_returns_tipheight()
+        {
+            Given(a_pow_node_with_api_enabled);
+            And(a_block_is_mined_creating_spendable_coins);
+            And(more_blocks_mined_past_maturity_of_original_block);
+            When(calling_getblockcount_via_api_returns_an_int);
```
Comment (majikandy) [inline src/Stratis.Bitcoin.IntegrationTests/API/ApiSpecification.cs]: second part of this sentence kinda weird, I think just `calling_getblockcount` is good enough

**Pull request merged**